napari/_vispy/overlays/axes.py

```diff
@@ -17,7 +17,6 @@ def __init__(self, *, viewer, overlay, parent=None) -> None:
         super().__init__(
             node=Axes(), viewer=viewer, overlay=overlay, parent=parent
         )
-        self.overlay.events.visible.connect(self._on_visible_change)
```

Comment:
This change is unrelated to the bug fix, but I picked it up along the way.

The base overlay already connects to the `visible` event of the overlay model. I'm not sure if the extra connection does any harm, but there's no reason for it.